### PR TITLE
Replace some usage of Azure/go-autorest

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Replaces some usage of the soon-to-be-unsupported [Azure/go-autorest](https://github.com/Azure/go-autorest) package.

**Which issue(s) this PR fixes**:

Refs #2974

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

File: azure/scope/machinepool.go [hunk 1]
```diff
@@ -377,11 +377,11 @@ func (m *MachinePoolScope) createMachine(ctx context.Context, machine azure.VMSS
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "scope.MachinePoolScope.createMachine")
 	defer done()
 
-	parsed, err := azureautorest.ParseResourceID(machine.ID)
+	parsed, err := arm.ParseResourceID(machine.ID)
```
Comment (jackfrancis) [inline azure/scope/machinepool.go:380]: Are we 100% confident these two `ParseResourceID` interfaces are identical?

Comment (mboersma) [reply azure/scope/machinepool.go:380]: The implementations are very different (see [go-autorest](https://github.com/Azure/go-autorest/blob/553a90ae65a6a2b18306fa04d7b1625960c5decb/autorest/azure/azure.go#L213) and [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/arm/internal/resource/resource_identifier.go#L63)), but the interface (API) and overall behavior appear to be the same–I tested with some resourceID strings. The SDKv2 `arm` version returns a different struct, but with the same information (although some field names changed).

Comment (jackfrancis) [reply azure/scope/machinepool.go:380]: It seems like we don't have existing UT cases for these transformations, so I'll defer to your validations, thank you for doing that work!

**Comment by CecileRobertMichon** [comment]
Flex passed but clusterclass failed...

/retest

**Comment by CecileRobertMichon** [comment]
>       -- Original Error: Code="PublicIPCountLimitReached" Message="Cannot create more
      than 100 public IP addresses for this subscription in this region." Details=[]'

/retest

**Comment by mboersma** [comment]
I don't think the `-optional` test will pass until #3322 merges, since it's the ClusterClass failure.

**Pull request merged**